### PR TITLE
Add possibility to group terms_by_org report by citation authority #2

### PR DESCRIPTION
**Pull request by josusky**
I am confused - GitHub tells me "Can’t automatically merge. Don’t worry, you can still create the pull request." 
But when I create a patch from my branch (in my fork) I can apply it on this master branch without error:
`git apply --check 0001-Add-possibility-to-group-terms_by_org-report-by-cita.patch`. 
I give up for today and will check it tomorrow if time permits.

### CODE REVIEW

**Comment by josusky** [comment]
Argh, it tries to pull 3 commits not just the last one.

File: pywiscat/wis1/catalogue.py [hunk 1]
```diff
@@ -80,3 +84,22 @@ def cache(ctx, directory, verbosity):
 
 
 catalogue.add_command(cache)
+
+
+@click.command()
+@click.pass_context
+@cli_callbacks
+@cli_option_directory
+@click.option('--term', '-t', 'terms', multiple=True, required=True)
```
Comment (tomkralidis) [inline pywiscat/wis1/catalogue.py]: Can we add a `help` arg to help describe the option?

File: pywiscat/wis1/util.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                LOGGER.debug(f'Found match #{match_count}, searched {processed}%')
+                matches.append(filename)
+            else:
+                LOGGER.debug(f'No match found, searched {processed}%')
+
+    LOGGER.debug(f'Found {len(matches)} matching metadata records.')
+    return matches
+
+
+def identifier(exml):
```
Comment (tomkralidis) [inline pywiscat/wis1/util.py:90]: Can we add annotations here (arg types, return type, docstring description of `:param exml:`

File: pywiscat/wis1/util.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    :returns: metadata record identifier
+    """
+
+    xpath = '//gmd:fileIdentifier/gco:CharacterString/text()'
+
+    return exml.xpath(xpath, namespaces=NAMESPACES)[0]
+
+
+def citation_authority(file_identifier: str):
```
Comment (tomkralidis) [inline pywiscat/wis1/util.py:104]: Can we add annotations here (return type, docstring description of `:param file_identifier:`

File: pywiscat/wis1/util.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            return components[3]
+    return ''
+
+
+def group_by_originator(file_list: list, group_by_authority: bool) -> dict:
+    """
+    Processes the given file list (MD records) and groups them by originator/pointOfContact
+    and optionally also by citation authority from the file identifier.
+
+    :param file_list: list of MD XML files
```
Comment (tomkralidis) [inline pywiscat/wis1/util.py:125]: add `:param group_by_authority:`

**Review by tomkralidis — COMMENTED** [review]
Minor changes suggested.  Can we also add example invocations in `README.md` ?  Thanks -- great work!

**Comment by tomkralidis** [comment]
@josusky I've tried to resolve the merge conflicts through the GitHub interface.  I'm +1 to merge following your confirmation.

**Comment by josusky** [comment]
Thanks, Tom.

**Pull request merged**